### PR TITLE
Enhancements in Validations and Comprehensive Testing

### PR DESCRIPTION
**Pull request by cancelei**
In this pull request, we've made significant strides in solidifying the robustness of our application. Our focus was on enhancing the model validations and expanding our test suite to cover edge cases and expected behaviors.

**Changes:**

**Validations:**

Introduced necessary validations for the User, Post, and Comment models. This ensures data integrity and consistency when saving to the database.
Added specific validations for the presence and length of certain attributes.

**Testing:**

Implemented RSpec tests covering both happy paths and edge cases for our models.
Expanded the test suite to ensure associations and methods within models work as intended.
Included specific tests for validations to make sure they trigger invalid data.

**Impact:**

These changes ensure that the application behaves predictably and data integrity is maintained. The comprehensive test suite provides a safety net, allowing for future enhancements or refactoring with confidence.

Please review the changes and provide feedback!

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (fobadara) [inline spec/models/user_spec.rb:1]: [REQUIRED] Kindly include a test for recent_post. It is a requirement to write tests for custom methods.

File: spec/models/post_spec.rb
Comment (fobadara) [inline spec/models/post_spec.rb:1]: [REQUIRED] Kindly include tests for recent_comments and update_user_posts_counter. It is a requirement for you to test all custom methods

File: spec/models/comment_spec.rb
Comment (fobadara) [inline spec/models/comment_spec.rb:1]: [REQUIRED] Kindly include a test for update_post_comment_counter. It is a requirement to test custom methods

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @cancelei, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good implementation of the principle of separating concerns.

## Required Changes ♻️
_Check the comments under the review._
I had to include some comments here because it couldn't be done inline
[REQUIRED] Kindly include the spec for the `like` model. I noticed that you don't have any tests for this model.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**